app/db/models/application_config.py

```diff
@@ -0,0 +1,195 @@
+# from __future__ import annotations
+
+import uuid
+from dataclasses import dataclass
+from enum import Enum
+from typing import List
+
+from flask_sqlalchemy.model import DefaultMeta
+from sqlalchemy import Column
+from sqlalchemy import ForeignKey
+from sqlalchemy import Integer
+from sqlalchemy import String
+from sqlalchemy.dialects.postgresql import ENUM
+from sqlalchemy.dialects.postgresql import JSON
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.ext.orderinglist import ordering_list
+from sqlalchemy.orm import Mapped
+from sqlalchemy.orm import mapped_column
+from sqlalchemy.orm import relationship
+from sqlalchemy.types import Boolean
+
+from app.db import db
+
+BaseModel: DefaultMeta = db.Model
+
+
+class ComponentType(Enum):
+    TEXT_FIELD = "TextField"
+    FREE_TEXT_FIELD = "FreeTextField"
+    EMAIL_ADDRESS_FIELD = "EmailAddressField"
+    UK_ADDRESS_FIELD = "UkAddressField"
+    HTML = "Html"
+    YES_NO_FIELD = "YesNoField"
+    RADIOS_FIELD = "RadiosField"
+
+
+@dataclass
+class Section(BaseModel):
+
+    round_id = Column(
+        "round_id",
+        UUID(as_uuid=True),
+        ForeignKey("round.round_id"),
+        nullable=True,  # will be null where this is a template and not linked to a round
+    )
+    section_id = Column(
+        "section_id",
+        UUID(as_uuid=True),
+        primary_key=True,
+        default=uuid.uuid4,
+    )
+    name_in_apply = Column("name_in_apply_json", JSON(none_as_null=True), nullable=False, unique=False)
+    template_name = Column("Template Name", String(), nullable=True)
+    is_template = Column("is_template", Boolean, default=False, nullable=False)
+    audit_info = Column("audit_info", JSON(none_as_null=True))
+    forms: Mapped[List["Form"]] = relationship(
+        "Form", order_by="Form.section_index", collection_class=ordering_list("section_index")
+    )
+    index = Column(Integer())
+    source_template_id = Column(UUID(as_uuid=True), nullable=True)
```

Comment:
Just thinking, anything below this (form/component) must also be a template, as 'copying' a template will require a copy of all the tree below. Do we need to ensure that if a form is a template and it is FK to a section that both are template? Happy to discuss if that isnt clear

Reply:
Good question. I think that will come up in the next ticket around cloning from a template so will work out any specifics like this then if that's ok?